packages/api/src/task/scheduler.ts

```diff
@@ -352,23 +352,24 @@ export class TaskScheduler {
     userId?: string,
     requesterId?: string,
   ) {
+    let uId = inputAsset?.userId || outputAsset?.userId || userId;
+    let user: User;
+    if (uId) {
+      user = await db.user.get(uId);
+    }
+    const projectId = inputAsset?.projectId || user?.defaultProjectId;
```

Comment:
Shouldn't the caller already send the user object and/or projectId? Haven't checked it deeply in the code but I feel like it will always be available (after all, the caller sends a `userId` here)

The `projectId` is less of a priority, but we should really avoid doing an extraneous DB query whenever possible (for the user). Every query is more DB load and more possibility of failure.